zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/util/AdvancedSearchFilterUtil.java

```diff
@@ -34,6 +35,8 @@ private static <T> T convertValue(final Class<T> tClass, final Object value) {
     if (!tClass.isInstance(value)) {
       if (tClass == String.class) {
         return (T) value.toString();
+      } else if (tClass == OffsetDateTime.class) {
+        return (T) OffsetDateTime.parse((String) value);
```

Comment:
❓ Just to clarify if I understand this correctly: this is trying to parse the input we get in the API for a date-time property, right? So, if I send value `"abc123"` to such an attribute, this will fail here? Is this tested in the `RestControllerTest` somehow? I can only see that we test with "valid" dates.

Reply:
Yes, this would fail with invalid date string. Good idea to test this too and probably needs a proper error message.
With other properties it's not a problem because there Jackson already parses it into the correct value (or fails).